tests/sqlite/sqlite.rs

```diff
@@ -745,7 +750,7 @@ async fn test_query_with_progress_handler() -> anyhow::Result<()> {
         Err(sqlx::Error::Database(err)) => assert_eq!(err.message(), String::from("interrupted")),
         _ => panic!("expected an interrupt"),
     }
-
+    assert!(CALLED.load(Ordering::Relaxed));
```

Comment:
Without these asserts, it's possible that these tests would still pass if the callback functions were never invoked. I went ahead and added these checks to the existing tests as well as the new tests that I created.